### ISSUE TITLE
OpenShift builder(s) for the operator nightly

### ISSUE THREAD

**Issue by vdemeester** [tektoncd/operator#333]
Because the `tektoncd/operator` has 2 targets (kubernetes and openshift), it would be nice to be able to run some tests against OpenShift as well as Kubernetes.

This should be possible by using a Pipeline with a provisionning task (using `openshift-install`) and a final task to destroy it afterwards.

We should start with the nightlies and think if it make sense to enable as an optional pipeline for PRs.

**Comment by vdemeester** [comment]
Another alternative would be to use `crc` inside of a job (prow, …) but this means we would need rather big node to execute our job on 😓

**Comment by nikhil-thomas** [comment]
yeah, CRC single node VM cluster sounds like a good starting point. 🧑‍💻 👍 
Then minimum [hardware requirements](https://access.redhat.com/documentation/en-us/red_hat_codeready_containers/1.29/html/getting_started_guide/installation_gsg#minimum-system-requirements-hardware_gsg) are:
```
CodeReady Containers requires the following system resources:

4 physical CPU cores
9 GB of free memory
35 GB of storage space
```

**Comment by vdemeester** [comment]
Going back on this, there is two options:
- we could work on provisionning openshift cluster (or crc) on the upstream tekton resources (using `openshift-install`)
  - pros:
    - we control everything
  - cons:
    - upstream pays for it
    - might be a bit brittle (at least it would be *way way way battle tested* than  [`openshift-ci`](https://github.com/openshift/release)
- we could hook up [`openshift-ci`](https://github.com/openshift/release) to this repository
  - pros:
    - Red Hat pays for it
    - it is battle tested
  - cons:
    - we control it less
    - we need to install an app and add a few accounts to the org. It's prow-based so there might be conflicts but I am hoping not.

@tektoncd/governing-board any preference over those options ? I would be "voting" for the later (using `openshift-ci`) as it is more stable and doesn't impact upstream budget.
